tests/Service/Parser/FunctionParserTest.php

```diff
@@ -157,12 +170,35 @@ public function shouldNotSupportWithoutQueryParameters(): void
      */
     public function shouldNotParseWithoutQueryParameters(): void
     {
-        $this->markTestSkipped('Skipped because there are no parsers yet');
         $this->expectException(InvalidArgumentException::class);
         $this->expectExceptionMessage('Query parameters must be set to FunctionParser.');
 
         foreach ($this->parser->parse('foo', 'bar') as $filter) {
             $this->fail('This should not be reached');
         }
     }
+
+    /**
+     * @test
+     */
+    public function shouldNotParseFunctionByExplicitValueDefinition(): void
+    {
+        // ?fun=fullName&firstName=Jon&surname=Snow
+        $queryParameters = ['fun' => 'fullName', 'firstName' => 'Jon', 'surname' => 'Snow'];
+
+        $this->parser->setQueryParameters($queryParameters);
+
+        foreach ($queryParameters as $column => $value) {
+            $this->assertTrue($this->parser->supports($column, $value));
+        }
+
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Explicit function definition by values must be an array of functions. fullName given.');
+
+        foreach ($queryParameters as $column => $value) {
+            foreach ($this->parser->parse($column, $value) as $filter) {
```

Comment:
This is because the `ExplicitFunctionDefinitionInValueParser` with the highest priority is first function parser called and its role is to find explicit function definition in value (_by column `fun`_) _anywhere_ in the queryParameters and parse it.

Btw there will be more parser with this kind of behaviour, since there are more filter function definitions, which might be in any order etc.